src/hooks/useBasisForm/validation/getInputRules.ts

```diff
@@ -0,0 +1,23 @@
+import { RegisterOptions } from "react-hook-form";
+import { InternalInputProps } from "../../../components/Input/types";
+import { inputRules } from "./validationRules";
+
+export const getInputRules = (props: Record<string, any>): RegisterOptions => {
+  const {
+    optional,
+    variant,
+  } = props as InternalInputProps;
+
+  let pattern = undefined;
+  if (variant === "decimal") {
+    pattern = inputRules.pattern.decimal;
+  }
+  if (variant === "numeric") {
+    pattern = inputRules.pattern.numeric;
+  }
+
+  return ({
+    required: optional || inputRules.required,
```

Comment:
Doesn't this mean that if the input is optional it will be required? Doesn't feel right to me. 🤔